core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -157,19 +161,27 @@ public TableBuilder withProperty(String key, String value) {
       return this;
     }
 
+    public TableMetadata newTableMetadata() {
```

Comment:
Extracting the logic that needs to be overridden for Nessie. 

Without changing this class, need to copy-paste a lot of code from here to NessieCatalog for overriding. So, code duplication will be more.